packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -457,6 +462,10 @@ class Environment {
   /// The [Source] value which is substituted with the path to [projectDir].
   static const String kProjectDirectory = '{PROJECT_DIR}';
 
+  /// The [Source] value which is substituted with the path to the parent of
+  /// [buildInfo.packageConfigPath].
+  static const String kWorkspaceDirectory = '{WORKSPACE_DIR}';
```

Comment:
so, in practice, `{WORKSPACE_DIR}` == `.dart_tool/`?

Reply:
Well - Should really be "grandparent" i guess. 
Tried to clarify